.github/workflows/libcxx-build-and-test.yaml

```diff
@@ -88,18 +88,22 @@ jobs:
           'generic-cxx20',
           'generic-cxx23'
         ]
-        cc: [ 'clang-20' ]
-        cxx: [ 'clang++-20' ]
+        cc: [ 'clang-21' ]
+        cxx: [ 'clang++-21' ]
         include:
           - config: 'generic-gcc-cxx11'
             cc: 'gcc-14'
             cxx: 'g++-14'
-          - config: 'generic-cxx23'
-            cc: 'clang-18'
-            cxx: 'clang++-18'
+          - config: 'generic-cxx26'
+            cc: 'clang-20'
+            cxx: 'clang++-20'
           - config: 'generic-cxx26'
             cc: 'clang-19'
             cxx: 'clang++-19'
+          # Release transition
+          - config: 'generic-cxx23'
+            cc: 'clang-18'
+            cxx: 'clang++-18'
```

Comment:
Do we really need this? I don't think there is much code that requires clang-18-specific workarounds currently and it's really not clear to me when we should drop it otherwise.

Reply:
Yes we need this. At the moment patches get actively backported to the release branch. Testing clang-18 enures that backporting is easy to do. Once LLVM 20.1.0 is released we can drop clang-18. (We've been doing this is the past too.)

Reply:
I agree with @mordante, this is easy to keep around for a few weeks and it reduces friction when backporting.

Reply:
I don't remember any rules on this and I though it was around for quite a while sometimes. If we agree that it can go after the initial release I'm fine with it, but I really don't want to keep it around for the whole cycle, since that would effectively extend our supported compilers by one version.